src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -49,18 +50,27 @@ export class AddCellAction extends Action {
 	) {
 		super(id, label, cssClass);
 	}
-	public async run(context: INotebookEditor): Promise<any> {
-		//Add Cell after current selected cell.
+	public async run(context: INotebookEditor | CellContext): Promise<any> {
 		let index = 0;
-		if (context && context.cells) {
-			let notebookcomponent = context as NotebookComponent;
-			let id = notebookcomponent.activeCellId;
-			if (id) {
-				index = context.cells.findIndex(cell => cell.id === id);
-				index = index + 1;
+		if (context instanceof CellContext) {
+			if (context && context.model && context.model.cells) {
+				let activeCellId = context.model.activeCell.id;
+				if (activeCellId) {
+					index = context.model.cells.findIndex(cell => cell.id === activeCellId) + 1;
+				}
+			}
+			context.model.addCell(this.cellType, index);
```

Comment:
I assume  model is expected to always be here - but currently you're checking that context.model exists in the if statement. But if it's undefined for some reason this will still attempt to access it and throw.